core/modules/widgets/checkbox.js

```diff
@@ -277,14 +296,19 @@ Selectively refreshes the widget if needed. Returns true if the widget or any of
 */
 CheckboxWidget.prototype.refresh = function(changedTiddlers) {
 	var changedAttributes = this.computeAttributes();
-	if(changedAttributes.tiddler || changedAttributes.tag || changedAttributes.invertTag || changedAttributes.field || changedAttributes.index || changedAttributes.listField || changedAttributes.filter || changedAttributes.checked || changedAttributes.unchecked || changedAttributes["default"] || changedAttributes["class"] || changedAttributes.disabled) {
+	if(changedAttributes.tiddler || changedAttributes.tag || changedAttributes.invertTag || changedAttributes.field || changedAttributes.index || changedAttributes.listField || changedAttributes.listIndex || changedAttributes.filter || changedAttributes.checked || changedAttributes.unchecked || changedAttributes["default"] || changedAttributes.indeterminate || changedAttributes["class"] || changedAttributes.disabled) {
 		this.refreshSelf();
 		return true;
 	} else {
 		var refreshed = false;
 		if(changedTiddlers[this.checkboxTitle]) {
 			var isChecked = this.getValue();
-			this.inputDomNode.checked = isChecked;
+			this.inputDomNode.checked = !!isChecked;
+			if(isChecked === undefined) {
+				this.inputDomNode.indeterminate = true;
+			} else {
+				this.inputDomNode.indeterminate = false;
+			}
```

Comment:
Question for whoever reviews this: would you prefer for that five-line `if` block to be collapsed down into a single line, as follows?

```js
this.inputDomNode.indeterminate = (isChecked === undefined);
```

Personally I would find that easier to read, but someone people would find it harder to read.

Reply:
We happily use constructions like  ` = !!isChecked;`, which is also cryptic for some programmers ... So I would be fine with the collapsed version, since it only contains 1 comparison. .. 

If the logic would be more complicated I'd prefer the long form. ... But it depends on @Jermolene ??

IMO if we collapse this one, we can also collapse some if's in the code above.

Reply:
I'm inclined to make one change per PR, i.e. if I'm going to change code that existed before the PR (like collapsing several of the `if` statements in the checkbox widget) then I think I'd prefer to do that in a separate PR. That way you avoid the situation where someone wants to merge change 1 from the PR but not change 2, and can't merge the PR as a whole because of that. So for now, I think I'll just collapse that one if-else block into a single line and leave the others alone until a later PR.

Reply:
Done in commit 3d04b8a.